specs/DesignDiscussions/InputValidation.md

```diff
@@ -0,0 +1,503 @@
+
+- By default we will validate on focus lost, so in a host with that behavior a user who has navigated to and away from an input will have seen the validation error, even if they subsequently hid it using a toggle or show card action. This means that in scenarios where the input is not required, but has a regex or other validation (such as the personalization example above) the user will already be aware that their field has not validated correctly. The scenario where a **required** input is not shown may be confusing to a user, but one could make the case that it's questionable card design to hide required fields behind toggle or show card actions.
+
+- If "All" doesn't work well for a particular  card, allowing the card author to explicitly specify inputs allows them to override this default behavior to configure the validation as appropriate for their scenario.
+
+##### Conclusion
+
+After talks with some of our MVPs, as in the case with the `validation behaviour`, we'll introduce this behaviour gradually, as for this time, the behavior for grabbing the inputs will be changed and will be made consistent across the different platforms. The new behaviour has to be consistent with the inclusion of input validation and as such we cannot validate inputs that are not visible as it may not provide a satisfactory card author or user experience, after all, if some inputs are in a hidden card, then probably the card author doesn't want to grab those inputs. At the same time, users must be able to know what input is validating incorrectly and we cannot simply take the decision of showing a hidden input as it may break the experience the card author devised.
+
+To explain the input retrieving behavior we landed on consider the following diagram where the main card A has two `ShowCard` actions.
+
+![img](assets/InputValidation/CardCascadeSample.png)
+
+The behaviour is as follows:
+* If the `Submit` action on card A is clicked, then only the inputs in card A are retrieved
+* If the `Submit` action on card B is clicked, then inputs in card A and card B are retrieved, but not the inputs in card C
+* If the `Submit` action on card C is clicked, then inputs in card A and card C are retrieved, but not the inputs in card B
+
+To summarize, all the inputs in the card where the `Submit` action was clicked and all the inputs in all the "parent" cards are retrieved, no inputs in "sibling" cards are retrieved. In the diagram above, card A would be a "parent" of B and C, and card B and C are siblings.
```

Comment:
>To summarize [](start = 0, length = 12)

What happens in a scenario where an input is hidden behind a toggle action? Are those retrieved regardless of whether or not they are hidden?